app/queries/outstanding_balance.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
```

Comment:
I like this pattern of pulling queries into their own classes/files.

Reply:
That's an abstraction I'd like to introduce. I've seen it in the Rails community for a long time (see 4 in https://codeclimate.com/blog/7-ways-to-decompose-fat-activerecord-models/) and although I used it before, I've been using it a lot lately, and IMO it has lots of pros.

some examples:
* https://medium.flatstack.com/query-object-in-ruby-on-rails-56ea434365f0
* https://thoughtbot.com/blog/active-record-eager-loading-with-query-objects-and-decorators
* https://thoughtbot.com/upcase/videos/advanced-query-optimization